c/authService.c

```diff
@@ -173,6 +174,202 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_WRONG_USER:
+          setResponseStatus(response,HTTP_STATUS_UNAUTHORIZED,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "The username entered is not found. Please try again.");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_NO_NEW_PASSSWORD:
+          setResponseStatus(response,HTTP_STATUS_BAD_REQUEST,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "No new password for expired password provided");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_WRONG_FORMAT:
+          setResponseStatus(response,HTTP_STATUS_BAD_REQUEST,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "The new password format is incorrect. Please try again.");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        case PASSWORD_RESET_TOO_MANY_ATTEMPTS:
+          setResponseStatus(response,HTTP_STATUS_TOO_MANY_REQUESTS,"ERROR");
+          setContentType(response,"text/json");
+          addStringHeader(response,"Server","jdmfws");
+          addStringHeader(response,"Transfer-Encoding","chunked");
+          addStringHeader(response, "Cache-control", "no-store");
+          addStringHeader(response, "Pragma", "no-cache");
+          writeHeader(response);
+
+          jsonStart(p);
+          jsonAddString(p, "status", "An incorrect password has been entered too many times and the account has been locked. Please contact your administrator for further instructions.");
+          jsonEnd(p);
+
+          finishResponse(response);
+          return HTTP_SERVICE_FAILED;
+        default:
+          setResponseStatus(response,HTTP_STATUS_BAD_REQUEST,"ERROR");
+          setContentType(response,"text/json");
```

Comment:
These headers are everywhere. You can greatly reduce duplication via this function: https://github.com/zowe/zowe-common-c/blob/master/c/httpserver.c#L4010